std/http/io.ts

```diff
@@ -340,43 +383,151 @@ export function parseHTTPVersion(vers: string): [number, number] {
   throw new Error(`malformed HTTP version ${vers}`);
 }
 
+/** Read HTTP/1.1 request line and */
 export async function readRequest(
   conn: Deno.Conn,
-  bufr: BufReader
+  opts?: {
+    r?: BufReader;
+    w?: BufWriter;
```

Comment:
I thought we wanted readRequest to not take a writer?

Reply:
Yes, it is needed to make request.w private and immutable. I agree that `readRequest` shouldn't take writer. But it's is impossible if we keep using class for ServerRequest.  Writer is needed only for responding. Here is one solution to split `ServerRequest` into request params and responder. Composing them should be done in `Server` after `readRequest.`

```ts
type Part1 = {
  url: string
  ...
}
type Part2 = {
  respond(res: ServerResponse)
}

type ServerRequest = Part1 & Part1;

function readRequest(r: Reader): Promise<Part1>

```

I think there are no reason to use class for `ServerRequest`.